progressReport/app.py

```diff
@@ -93,10 +117,31 @@ def assign_node_levels(node, student_levels_count, class_levels_count):
             node["class_level"] = sum(children_class_levels) // len(children_class_levels)
         return node["student_level"], node["class_level"]
 
-    school_name = request.args.get("school", "Berkeley")
-    course_name = request.args.get("class", "CS10")
-    student_mastery = request.args.get("student_mastery", "000000")
-    class_mastery = request.args.get("class_mastery", "")
+    # These values will be initialized in different ways depending on whether this route receives a GET or POST request
+    school_name = "Berkeley"
+    course_name = "CS10"
+    student_mastery = "000000"
+    class_mastery = ""
+
+    # The plan can be that the POST request contains all info we currently relay to CM in the url.
+    if request.method == "POST":
+        request_as_json = request.get_json()
+        validate_mastery_learning_post_request(request_as_json)
+        school_name = request_as_json["school"]
+        course_name = request_as_json["class"]
+        class_mastery = request_as_json["class_mastery"]
+
+        # To be consistent with the existing code, we can continue using a student_mastery string, which we can construct a student_mastery string using the content of the POST request.
+        student_mastery = ""
```

Comment:
@Connor-Bernard , I have a request about the structure of the POST request. Could GV send the student_mastery string as a key-value pair in the POST instead of key-value pairs for every child-concept? This would simplify implementation for CM, and allow us to use existing code with fewer changes. If this can’t be done, no worries! The below code constructs the student_mastery string

Reply:
Unfortunately not.  The student_mastery string has positional (undefined) args.  This is actually the exact reason why we are moving to a structured approach.

Reply:
Understood!